dot/digest/digest.go

```diff
@@ -48,7 +48,6 @@ type Handler struct {
 
 	// block notification channels
 	imported    chan *types.Block
```

Comment:
maybe store this as `<-chan *types.Block` so it's clear it's read-only. same for other services

Reply:
I'm not able to store this as `<-chan *types.Block` because treating this as a receive channel breaks other parts of the code.